fe1-web/src/features/evoting/network/__tests__/ElectionHandler.test.ts

```
@@ -104,31 +229,151 @@ describe('ElectionHandler', () => {
+      });
+
+      expect(storedElection.electionStatus).toEqual(ElectionStatus.NOT_STARTED);
+
+      expect(
+        handleElectionOpenMessage(
+          getEventFromId,
+          updateEvent,
+        )({
+          ...mockMessageData,
+          messageData: new OpenElection({
+            lao: mockLaoIdHash,
+            election: mockElectionId,
+            opened_at: TIMESTAMP,
+          }),
+        }),
+      ).toBeTrue();
+
+      // no warning should have been printed
+      expect(warn).toHaveBeenCalledTimes(0);
+
+      // check whether getEventFromId has been called correctly
+      expect(getEventFromId.mock.calls[0][0]).toEqual(mockElectionId);
+      expect(getEventFromId).toHaveBeenCalledTimes(1);
+
+      // check whether updateEvent has been called correctly
+      expect(updateEvent.mock.calls[0][0]).toEqual(mockLaoIdHash);
+      expect(updateEvent.mock.calls[0][1]).toHaveProperty(
+        'id',
+        mockElectionNotStarted.id.valueOf(),
+      );
+      expect(updateEvent.mock.calls[0][1]).toHaveProperty('electionStatus', ElectionStatus.OPENED);
+      expect(updateEvent).toHaveBeenCalledTimes(1);
+
+      // check if the status was changed correctly
+      expect(storedElection.electionStatus).toEqual(ElectionStatus.OPENED);
+    });
+  });
+
+  describe('election#castVote', () => {
+    it('should return false if the object is not "election"', () => {
+      expect(
+        handleCastVoteMessage(
+          getMockLao,
+          getEventFromIdDummy,
+          updateEventDummy,
+        )({
+          ...mockMessageData,
+          messageData: {
+            object: ObjectType.CHIRP,
+            action: ActionType.CAST_VOTE,
           },
         }),
       ).toBeFalse();
 
       expect(warn).toHaveBeenCalledTimes(1);
       // check if the printed warning message contains substring
-      expect(warn.mock.calls[0][0]).toMatch(/LAO/i);
+      expect(warn.mock.calls[0][0]).toMatch(/unsupported message/i);
```

This kind of logic makes me think that the API is wrong.
We should probably throw exceptions instead of returning true/false.
This, however, would be a rather bigger change to do
Definitely outside the scope of this PR.